### Data Set: Tokyo
Visualización de datos del data set Tokyo.</br>
Exportación de librerias de Visualización del data set:

In [43]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
#plt.style.use("seaborn")

Visualizamos las primeras 10 Columnas del dataset, y vemos que datos tenemos

In [44]:
df = pd.read_csv("../data/tokyo_airbnb.csv")
df.head(10)

id                                               name  host_id  \
0    35303  La Casa Gaienmae C Harajuku, Omotesando is nearby   151977   
1   197677                          Oshiage Holiday Apartment   964081   
2   289597                  Private apt in central Tokyo #203   341577   
3   370759  Cozy flat #203, local area YET 10 mins to shib...  1573631   
4   700253                  Private apt in central Tokyo #201   341577   
5   776070                             Kero-kero house room 1   801494   
6   872235                 RELAX AT HOME:Shimo 5'/Shibuya 10'  1573631   
7   899003                  Classy room @Shinjuku, Takadanoba  4799233   
8   905944           4F - Near Shinjuku & Shibuya w/Free WiFi  4847803   
9  1007534                      Kichijoji Parkside Attic Room  5068769   

                   host_name  neighbourhood_group neighbourhood  latitude  \
0                     Miyuki                  NaN    Shibuya Ku  35.67152   
1            Yoshimi & Marek                  NaN     Sumida Ku  35.71721   
2                   Hide&Kei                  NaN     Nerima Ku  35.74267   
3        Gilles,Mayumi,Taiki                  NaN   Setagaya Ku  35.66344   
4                   Hide&Kei                  NaN     Nerima Ku  35.74264   
5  Kotaro Kei And Litte Miya                  NaN    Arakawa Ku  35.73818   
6        Gilles,Mayumi,Taiki                  NaN   Setagaya Ku  35.66832   
7                         Yu                  NaN   Shinjuku Ku  35.70865   
8        Best Stay In Tokyo!                  NaN    Shibuya Ku  35.67968   
9                   Caroline                  NaN    Mitaka Shi  35.69952   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  139.71203     Private room   4196              28                 18   
1  139.82596  Entire home/apt  10975               3                156   
2  139.65810  Entire home/apt   4196              30                107   
3  139.65593  Entire home/apt   6994              29                 99   
4  139.65832  Entire home/apt   3981              30                101   
5  139.77009     Private room   6456               3                215   
6  139.65159  Entire home/apt   9469              29                 41   
7  139.69681  Entire home/apt   5165              30                 90   
8  139.67949     Private room   9684               3                175   
9  139.57600     Private room   6564               3                 85   

  last_review  reviews_per_month  
0  2018-07-28               0.20  
1  2019-05-21               1.65  
2  2019-05-25               1.25  
3  2019-05-07               1.55  
4  2019-05-03               1.24  
5  2019-06-16               2.68  
6  2018-08-20               0.54  
7  2019-06-11               1.16  
8  2019-06-19               2.31  
9  2019-06-11               2.65

Aplicamos una función para saber: 
Dataset tiene 14 columnas y 11466 fila.
- Valores únicos
- Ausencia de valores 
- Duplicado
- Tipo de valores

In [45]:
def summary(df):
    total_values = df.shape[0]
    missing_data = df.isnull().sum()
    missing_percentage = (missing_data / total_values) * 100
    print(f"Dataset has {df.shape[1]} features and {df.shape[0]} rows.")
    summary = pd.DataFrame(index=df.columns)
    summary["Unique"] = df.nunique().values
    summary["Missing"] = df.isnull().sum().values
    summary['Missing %'] = ((missing_data / total_values) * 100).round(2)
    summary["Duplicated"] = df.duplicated().sum()
    summary["Types"] = df.dtypes
    return summary

summary(df)

Dataset has 14 features and 11466 rows.


Unique  Missing  Missing %  Duplicated    Types
id                    11466        0       0.00           0    int64
name                  10810        0       0.00           0   object
host_id                2954        0       0.00           0    int64
host_name              2054       16       0.14           0   object
neighbourhood_group       0    11466     100.00           0  float64
neighbourhood            56        0       0.00           0   object
latitude               7295        0       0.00           0  float64
longitude              8140        0       0.00           0  float64
room_type                 3        0       0.00           0   object
price                   371        0       0.00           0    int64
minimum_nights           30        0       0.00           0    int64
number_of_reviews       251        0       0.00           0    int64
last_review             413     1677      14.63           0   object
reviews_per_month       692     1677      14.63           0  float64

Hayamos el porcentaje de peso en nulos/ausencia del dataset. 
Observamos que la columna neighbourhood_group tiene un 100% de ausencia de datos.
Las columnas: 
last_review             14.63% 
reviews_per_month       14.63%

Veremos más adelante que problemática nos pueden surgir y si son útiles 

In [46]:
df_clean = df.drop(['id', 'neighbourhood_group', 'latitude', 'longitude', 'last_review', 'reviews_per_month'], axis=1, inplace=True)